actionpack/lib/action_dispatch/routing/mapper.rb

```diff
@@ -231,12 +231,21 @@ def normalize_options!(options, path_params, modyoule)
               if to.nil?
                 controller = default_controller
                 action = default_action
-              elsif to.is_a?(String) && to.include?("#")
-                to_endpoint = to.split("#").map!(&:-@)
-                controller  = to_endpoint[0]
-                action      = to_endpoint[1]
               else
-                raise ArgumentError, ":to must respond to `action` or `call`, or it must be a String that includes '#'"
+                if to.is_a?(String)
+                  controller_or_action, action = to.split("#", 2).map!(&:-@)
```

Comment:
Specifying the numbers of splits here makes sure we handle the `"controller#"` case correctly (and that we don't assume `controller` is the action).